### PR TITLE
Draft: Bazel recursive stubgen example

### PR DESCRIPTION
**Pull request by cemlyn007**
I made this branch mainly for another [MR](https://github.com/nicholasjng/nanobind-bazel/pull/44) I am trying to merge with the help of @nicholasjng. If/when the nanobind-bazel MR can land, would you maintainers want this example merged as well?

It's meant to be a slight modification of the existing example to create an example sub-module and makes a small modification to the `setup.py` to copy files whilst preserving the module structure.

### CODE REVIEW

File: src/nanobind_example/sub_ext/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from ..nanobind_example_ext.sub_ext import sub
```
Comment (nicholasjng) [inline src/nanobind_example/sub_ext/__init__.py:1]: This means that all other submodules reference the top-level nanobind extension, right?

So if I do `sm.def_submodule("div_ext", "A submodule for dividing integers");` I need to import `from ...nanobind_example_ext.sub_ext.div_ext import div` in `src/nanobind_example/sub_ext/div_ext/__init__.py`?

File: setup.py [hunk 2]
```diff
@@ -102,7 +102,10 @@ def bazel_build(self, ext: BazelExtension) -> None:
                     should_copy = True
 
                 if should_copy:
-                    shutil.copyfile(root / fp, libdir / fp)
+                    dstdir = libdir / os.path.relpath(root, srcdir)
```
Comment (nicholasjng) [inline setup.py:106]: I'm starting to think I should just spend some time on a file inclusion function compatible with `shutil.copytree()`. Thanks for the fix!

**Review by nicholasjng — COMMENTED** [review]
Thanks for the addition! I think this is very valuable to have for nanobind-bazel.

I actually have write access to this repo, since I sometimes need to update the `bazel` branch when configurations change (or setuptools breaks again). If @wjakob agrees, I can take over all Bazel-related PRs on this repo.

**Comment by nicholasjng** [comment]
Hey @cemlyn007 , since rules_python just released v1.0.0 with a fix for the Windows library locations, I updated the setup.py (and some other files) accordingly to test if we can build nanobind on Windows with only a hermetic interpreter.

Could I ask you to rebase on current `bazel`? Then I'll merge this right after.

**Comment by wjakob** [comment]
> If @wjakob agrees, I can take over all Bazel-related PRs on this repo.

I agree.

A tangent : would it make sense to advertise more broadly that there is a `bazel` branch on the `master` branch? This may not be obvious to visitors.

**Comment by nicholasjng** [comment]
Thanks! And sure, I can add a note to the readme or something similar.

**Comment by nicholasjng** [comment]
Note: I declared the dependency on `nanobind_bazel` to be against a stable release from BCR, since that is how most users will use the package. Instead of keeping the `local_path_override` in this repo, I will add it in nanobind-bazel CI.

Currently CI is all red, since we're passing arguments to stubgen that have not been implemented yet (PR is open already). Once nanobind-bazel v2.4.0 is out on the BCR, this will be fixed.

**Pull request merged**